src/leveled_log.erl

```diff
@@ -458,30 +462,11 @@ log_randomtimer(LogReference, Subs, StartTime, RandomProb) ->
             ok
     end.
 
-localtime_ms() ->
-    {_, _, Micro} = Now = os:timestamp(),
-    {Date, {Hours, Minutes, Seconds}} = calendar:now_to_local_time(Now),
-    {Date, {Hours, Minutes, Seconds, Micro div 1000 rem 1000}}.
-
--spec log_prefix(
-    tuple(), atom(), atom(), non_neg_integer(), pid()) -> io_lib:chars().
-log_prefix({{Y, M, D}, {H, Mi, S, Ms}}, LogLevel, LogRef, DBid, Pid) ->
-    [integer_to_list(Y), $-, i2l(M), $-, i2l(D),
-    $T, i2l(H), $:, i2l(Mi), $:, i2l(S), $., i3l(Ms),
-    " log_level=", atom_to_list(LogLevel), " log_ref=", atom_to_list(LogRef),
-    " db_id=", integer_to_list(DBid), " pid=", pid_to_list(Pid), " "].
-
--spec i2l(non_neg_integer()) -> list().
-i2l(I) when I < 10 ->
-    [$0, $0+I];
-i2l(I) ->
-    integer_to_list(I).
-
--spec i3l(non_neg_integer()) -> list().
-i3l(I) when I < 100 ->
-    [$0 | i2l(I)];
-i3l(I) ->
-    integer_to_list(I).
+-spec log_prefix(atom(), non_neg_integer(), pid()) -> io_lib:chars().
```

Comment:
Creating a list for strings is perfectly fine. Just want to point out that using the template functionality build in in `logger` is an alternative.

https://www.erlang.org/doc/apps/kernel/logger_formatter.html#t:template/0